### PR TITLE
Refactor script - WIP

### PR DESCRIPTION
**Pull request by sourdzl**
not ready for review/land.

breaks out georgios' script into sub components, to prepare for upgrades in functionality

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+clap = {version = "4.0.25", features = ["derive"]}
 ethers = "1.0.1"
 ethers-flashbots = "0.11.0"
 eyre = "0.6.8"
 fastrand = "1.8.0"
-tokio = { version = "1.21.2", features = ["macros"] }
+tokio = {version = "1.21.2", features = ["macros"]}
 tracing = "0.1.37"
-tracing-subscriber = { version = "0.3.16", features = ["env-filter"] }
+tracing-subscriber = {version = "0.3.16", features = ["env-filter"]}
```
Comment (gakonst) [inline Cargo.toml]: remove the formatting changes

File: src/bundle_builder.rs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        .data(blob)
+        .gas_price(gas_price);
+
+    return tx;
+}
+
+fn generate_random_data(size: u32) -> Vec<u8> {
+    let blob = rand::thread_rng()
+        .sample_iter(Standard)
+        .take(6 * 1024)
```
Comment (gakonst) [inline src/bundle_builder.rs]: size unused

File: src/bundle_builder.rs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+const KB: u32 = 1024;
+
+/// Arbitrarily chosen number to cover for nonce+from+to+gas price size in a serialized
+/// transaction.  TODO: get the actual overhead from the signing, etc. to pack more fully
+const TRIM_BYTES: u32 = 500;
+
+#[tracing::instrument(skip_all, name = "construct_bundle")]
+fn construct_tx(
+    chain_id: u64,
+    address: <Wallet<ethers::core::k256::ecdsa::SigningKey> as Trait>::SignerMiddleware,
```
Comment (gakonst) [inline src/bundle_builder.rs]: Make `Address` not a signer middleware

File: src/mev_boost_tools.rs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+) -> (
+    H160,
+    U256,
+    u64,
+    Arc<
+        SignerMiddleware<
+            FlashbotsMiddleware<Arc<Provider<Provider>>, Wallet<SigningKey>>,
+            Wallet<SigningKey>,
+        >,
+    >,
```
Comment (gakonst) [inline src/mev_boost_tools.rs]: You might be able to call this `impl Middleware` in the return type, but OK to leave as-is rn.

File: src/bundle_builder.rs (hunk 3 above)
Comment (sourdzl) [reply src/bundle_builder.rs]: thank you ser

File: src/bundle_builder.rs (hunk 2 above)
Comment (sourdzl) [reply src/bundle_builder.rs]: thaanks

File: src/mev_boost_tools.rs (hunk 4 above)
Comment (sourdzl) [reply src/mev_boost_tools.rs]: thanks - tried this but didnt get it working, so kept in the long way.  but i added an `<Http>` on the inner provider type

File: src/bundle_builder.rs [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        .from(address)
+        .to(receiver)
+        .data(blob)
+        .gas_price(gas_price);
+
+    return tx;
+}
+
+async fn get_signed_tx(chain_id: u64,
+address: H160, receiver: H160, chunk:usize, gas_price: U256, provider, nonce: U256)-> Result<Bytes, WalletError> {
```
Comment (loren) [inline src/bundle_builder.rs]: `provider:&Arc<SignerMiddleware<M, LocalWallet>>` ?

Comment (sourdzl) [reply src/bundle_builder.rs]: ohh thanks lemem try that.  simpler than copy pasta the giant thing around.

Comment (sourdzl) [reply src/bundle_builder.rs]:
```
cannot find type `M` in this scope
not found in this scoperustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message?2#file%3A%2F%2F%2FUsers%2Fdanlee%2Fcode%2Fstress4844%2Fsrc%2Fbundle_builder.rs)
bundle_builder.rs(38, 23): you might be missing a type parameter: `<M>`
```
hmm not used to generics in rust

File: Cargo.toml [hunk 6]
```diff
@@ -1,17 +1,19 @@
 [package]
+edition = "2021"
```
Comment (sourdzl) [inline Cargo.toml:2]: todo: revert

File: src/bundle_builder.rs [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+    let tx = tx.into();
+    let sender = provider.default_sender().unwrap_or_default();
+    let signature = provider.sign_transaction(&tx, sender).await?;
+    tracing::debug!("sender is {sender} , signature = {signature}");
+    let rlp = tx.rlp_signed(&signature);
+
+    println!("{}", serde_json::to_string(&tx)?);
+
+    // ad hoc test: submit directly
+    let tx = provider.send_transaction(tx, None).await?.await?;
```
Comment (sourdzl) [inline src/bundle_builder.rs:69]: added this line because bundle submission was giving 

`Error: Client error: {"error":"unable to decode txs"}`

Comment (sourdzl) [reply src/bundle_builder.rs:69]: need to figure out whats wrong with the generated transaction, will compare against master

**Pull request closed without merging**